core/src/components/button/button.ionic.scss

```diff
@@ -1,32 +1,55 @@
 @use "../../themes/ionic/ionic.globals.scss" as globals;
-@import "./button";
```

Comment:
Please revert this. We want to be consistent, we import the base styles to all components for all themes. If we want to avoid values being mixed, then we would use the variables to override any values as shown in other themes.

Reply:
Consistency is the reason I did this change. We were mixing up not only themes, but Design Systems. 

The problem here is that the current base styles (button.scss) are based on the current Design System on Ionic Framework, that supports two themes: ios/md.

We are introducing a new Design System, and a new theme (Ionic Theme) to the Framework. And I believe we should aim to keep them separated as much as possible, until its possible to join them in the future when/if we decide to update the md/ios for the new Design System (in the meantime, it might mean a bit of duplicate code on the framework, but the output for CSS for each theme will be cleaner).

I thought we were all on board with this, as it was discussed/reviewed and accepted on this [CSS Architecture Design Doc](https://outsystemsrd.atlassian.net/wiki/spaces/RDMBVS/pages/3982655893/CSS+Architecture+-+New+Design+System), particularly [this page about scss](https://outsystemsrd.atlassian.net/wiki/spaces/RDMBVS/pages/3991109633/SCSS+Architecture).

My PR goes in the direction of meeting the architecture represented on this diagram:
![image](https://github.com/ionic-team/ionic-framework/assets/32780808/ac8cc258-0156-40f2-897e-314e9466fce3)

But if you're not comfortable with it and think we are doing something wrong here, I would say we can schedule a team meeting to discuss this general architecture stuff., so that everyone as a more active way to give their insights about it. After that we can come back to this PR, and change it, accept it or just close it 👍